src/openrct2-ui/windows/Ride.cpp

```diff
@@ -4238,6 +4242,13 @@ static void window_ride_colour_mouseup(rct_window* w, rct_widgetindex widgetInde
         case WIDX_PAINT_INDIVIDUAL_AREA:
             tool_set(w, WIDX_PAINT_INDIVIDUAL_AREA, Tool::PaintDown);
             break;
+        case WIDX_SELL_ITEM_RANDOM_COLOUR_CHECKBOX:
+            auto ride = get_ride(w->number);
```

Comment:
To make this work in multiplayer the change will have to go through the game action framework. I think you should be able to extend RideSetAppearanceAction to add the random setting.